scripts/pylib/twister/twisterlib/testinstance.py

```diff
@@ -270,9 +270,11 @@ def calculate_sizes(self, from_buildlog: bool = False, generate_warning: bool =
     def get_elf_file(self) -> str:
         fns = glob.glob(os.path.join(self.build_dir, "zephyr", "*.elf"))
         fns.extend(glob.glob(os.path.join(self.build_dir, "zephyr", "*.exe")))
-        fns = [x for x in fns if '_pre' not in os.path.split(x)[-1]]
-        # EFI elf files
-        fns = [x for x in fns if 'zefi' not in os.path.split(x)[-1]]
+        blocklist = [
+                'remapped', # used for xtensa plaforms
+                'zefi', # EFI for Zephyr
+                '_pre' ]
+        fns = [x for x in fns if not any(bad in os.path.split(x)[-1] for bad in blocklist)]
```

Comment:
There are already a few lines of code parsing `.config` in `twisterlib/runners.py#FilterBuilder`, maybe they could be promoted to a new `get_config()` function and re-used here:

```python

   bin_name = get_config()["CONFIG_KERNEL_BIN_NAME"]
   if os.path.exists(f"{bin_name}.elf"):
     ...
   elif os.path.exists(f"{bin_name}.exe"):
    ...
   else:
      raise BuildError(...)
```